lib/truffle/truffle/patching.rb

```diff
@@ -0,0 +1,38 @@
+module Truffle::Patching
+  TRUFFLE_PATCHES_DIRECTORY = File.join(Truffle::Boot.ruby_home, 'lib', 'patches')
+  TRUFFLE_PATCHES           = Dir.glob(File.join(TRUFFLE_PATCHES_DIRECTORY, '**', '*.rb')).
+      select { |path| File.file? path }.
+      map { |path| path[(TRUFFLE_PATCHES_DIRECTORY.size+1)..-4] }.
+      reduce ({}) { |h, v| h.update v => true }
```

Comment:
I find the pure functional approach here more verbose and also much less efficient.
I propose:
```ruby
TRUFFLE_PATCHES = {}
Dir.glob("#{TRUFFLE_PATCHES_DIRECTORY}/**/*.rb") do |path|
  if File.file? path
    feature = path[(TRUFFLE_PATCHES_DIRECTORY.size+1)...-3]
    TRUFFLE_PATCHES[feature] = true
  end
end
```